### ISSUE TITLE
release_candidate broken

### ISSUE THREAD

**Issue by pierrethibault** [ptycho/ptypy#141]
Something having to do with recent commits.
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-17-7e5d8b111290> in <module>()
----> 1 import ptypy

/home/thibault/.local/lib/python2.7/site-packages/ptypy/__init__.py in <module>()
     54 from . import utils
     55 from . import io
---> 56 from . import core
     57 from . import simulations
     58 from . import resources

/home/thibault/.local/lib/python2.7/site-packages/ptypy/core/__init__.py in <module>()
      9 """
     10 from .classes import *
---> 11 from .ptycho import *
     12 from .manager import *
     13 from . import xy

/home/thibault/.local/lib/python2.7/site-packages/ptypy/core/ptycho.py in <module>()
     16 from ..utils.verbose import logger, _, report, headerline, log
     17 from ..utils import parallel
---> 18 from .. import engines
     19 from ..io import interaction
     20 from .classes import Base, Container, Storage, PTYCHO_PREFIX

/home/thibault/.local/lib/python2.7/site-packages/ptypy/engines/__init__.py in <module>()
     13 from .. import utils as u
     14 from base import BaseEngine, DEFAULT_iter_info
---> 15 from . import DM
     16 from . import DM_simple
     17 from . import ML

/home/thibault/.local/lib/python2.7/site-packages/ptypy/engines/DM.py in <module>()
     16 from . import BaseEngine
     17 from ..utils.descriptor import defaults_tree
---> 18 from ..core.manager import Full, Vanilla
     19 
     20 __all__ = ['DM']

/home/thibault/.local/lib/python2.7/site-packages/ptypy/core/manager.py in <module>()
     16 import numpy as np
     17 from collections import OrderedDict
---> 18 from . import illumination
     19 from . import sample
     20 from . import geometry

/home/thibault/.local/lib/python2.7/site-packages/ptypy/core/illumination.py in <module>()
    198 
    199 DEFAULT = illumination_desc.make_default(99)
--> 200 DEFAULT_aperture = DEFAULT.aperture
    201 
    202 __all__ = ['init_storage', 'aperture', 'DEFAULT']

/home/thibault/.local/lib/python2.7/site-packages/ptypy/utils/parameters.pyc in __getattr__(self, name)
    102             return self.__getitem__(name)
    103         except KeyError as ke:
--> 104             raise AttributeError(ke)
    105 
    106     __setattr__ = __setitem__

AttributeError: 'aperture'
```

**Comment by aaron-parsons** [comment]
Right I figured it out. Firstly, the statement is only true in the case of bools. So I added that and now the utils import. Now we have errors because actually what we want in the case of the parameter being true is to instantiate all the sub-parameters. so autoplot.threaded doesn't exist

In the old version of ptypy this used to be already done as a default which we would then update. The sub-parameters are not currently instantiated anywhere unless we actually pass them in via the script. It seems (at a brief look) this is also true for other parameters which behave in the same way.
